src/test/java/org/broadinstitute/hellbender/tools/dataflow/pipelines/MarkDuplicatesDataflowIntegrationTest.java

```diff
@@ -1,37 +1,78 @@
 package org.broadinstitute.hellbender.tools.dataflow.pipelines;
 
+import com.google.common.collect.ImmutableList;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.SAMReadGroupRecord;
 import org.broadinstitute.hellbender.CommandLineProgramTest;
 import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
 import org.broadinstitute.hellbender.engine.ReadsDataSource;
 import org.broadinstitute.hellbender.tools.picard.sam.markduplicates.MarkDuplicatesIntegrationTest;
+import org.broadinstitute.hellbender.utils.read.markduplicates.DuplicationMetrics;
 import org.broadinstitute.hellbender.utils.read.GATKRead;
 import org.broadinstitute.hellbender.utils.test.ArgumentsBuilder;
 import org.testng.Assert;
 import org.testng.annotations.DataProvider;
+import org.testng.annotations.BeforeClass;
 import org.testng.annotations.Test;
 
 import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
 import java.io.IOException;
+import java.util.List;
 
+public class MarkDuplicatesDataflowIntegrationTest extends CommandLineProgramTest {
+    protected static String TEST_BASE_NAME = null;
 
-public class MarkDuplicatesDataflowIntegrationTest extends CommandLineProgramTest{
+    @BeforeClass
+    public void setUp() {
+        TEST_BASE_NAME = "MarkDuplicatesDataflow";
+    }
 
     @DataProvider(name = "md")
     public Object[][] md(){
         return new Object[][]{
-                {new File(MarkDuplicatesIntegrationTest.TEST_DATA_DIR,"example.chr1.1-1K.unmarkedDups.noDups.bam"), 20, 0},
-                {new File(MarkDuplicatesIntegrationTest.TEST_DATA_DIR,"example.chr1.1-1K.unmarkedDups.bam"), 90, 6},
-                {new File(MarkDuplicatesIntegrationTest.TEST_DATA_DIR,"example.chr1.1-1K.markedDups.bam"), 90, 6},  //90 total reads, 6 dups
+            // The first two values are total reads and duplicate reads. The list is an encoding of the metrics
+            // file output by this bam file. These metrics files all match the outputs of picard mark duplicates.
+            {new File(MarkDuplicatesIntegrationTest.TEST_DATA_DIR,"example.chr1.1-1K.unmarkedDups.noDups.bam"), 20, 0,
+             ImmutableList.of(ImmutableList.of("Solexa-16419", 0L, 3L, 0L, 0L, 0L, 0L, 0.0, 0L),
+                              ImmutableList.of("Solexa-16416", 0L, 1L, 0L, 0L, 0L, 0L, 0.0, 0L),
+                              ImmutableList.of("Solexa-16404", 0L, 3L, 0L, 0L, 0L, 0L, 0.0, 0L),
+                              ImmutableList.of("Solexa-16406", 0L, 1L, 0L, 0L, 0L, 0L, 0.0, 0L),
+                              ImmutableList.of("Solexa-16412", 0L, 1L, 0L, 0L, 0L, 0L, 0.0, 0L))},
+            {new File(MarkDuplicatesIntegrationTest.TEST_DATA_DIR,"example.chr1.1-1K.unmarkedDups.bam"), 90, 6,
+             ImmutableList.of(ImmutableList.of("Solexa-16419", 4L, 4L, 4L, 0L, 0L, 0L, 0.0, 0L),
```

Comment:
Potentially, but I did like having all the data just listed here and easy to look at when comparing to the metrics file output by the non-dataflow version. Also, wasn't sure metricsfile had the right constructor.